### ISSUE TITLE
Text to number coercion doesn't work correctly

### ISSUE THREAD

**Issue by xxASZxx** [ClosedXML/ClosedXML#1891]
## Read and complete the full issue template

Do not randomly delete sections. They are here for a reason.

**Do you want to request a *feature* or report a *bug*?**
- [] Bug
- [ ] Feature
- [X] Question

**Did you test against the latest CI build?**
- [X ] Yes
- [ ] No

If you answered `No`, please test with [the latest development build](https://ci.appveyor.com/project/ClosedXML/ClosedXML/branch/develop/artifacts) first.

**Version of ClosedXML**

e.g. 0.96

**What is the current behavior?**

Hello.

Can you tell me how to recalculate the filter? CTRL + ALT + L
The code is below and tried other different ways but when it doesn't do the recalculation.

I enter the data in the first sheet and then I need to recalculate and save the changes.

**Code to reproduce problem:**
```c#
public void Main()
{
    var _workbook = new XLWorkbook(pathFile);
            var worckSheet = _workbook.Worksheet(2);

            for (int l = 2; l < _workbook.Worksheets.Count; l++)
            {
                _workbook.Worksheet(l).AutoFilter.Reapply();
            }

            _workbook.Save();
}
```

- [x] I attached a sample spreadsheet.  (You can drag files on to this issue)
[Orig — копия.xlsx](https://github.com/ClosedXML/ClosedXML/files/9866837/Orig.xlsx)

**Comment by jahav** [comment]
I don't think this will work at this time. 

There are at least two issues (possibly more):
* The column C in the first worksheet is a shared string with a value of dates. Although format is 14 (date), ClosedXml interprets it as a text. Thus the J is not calculated correctly (text - date = error -> this value should be number)
* AutoFilter conditions are limited and in this case they don't. They should probably use standard excel formula semantic, though I am not sure.

I tried to overcome first problem by setting explicitely DataType to DateTime, but that failed later in AutoFilter.

**Comment by jahav** [comment]
Simplified example:
```csharp
using var wb = new XLWorkbook(@"DateNotRepresentedAsNumber.xlsx");
var ws = wb.Worksheets.Worksheet(1);
wb.RecalculateAllFormulas();
Assert.AreEqual(87700d, ws.Cell(1, 2).Value); // ASIS XLError.IncompatibleValue because text * 2 doesn't make sense
```
[DateNotRepresentedAsNumber.xlsx](https://github.com/ClosedXML/ClosedXML/files/9874043/DateNotRepresentedAsNumber.xlsx)

**Comment by jahav** [comment]
I have looked into it a little more and there is a better explanation, formula engine should be able to coerce a string in a date form to number (e.g. formula "2020-03-20"*2).  It is also how excel shows the formula is evaluated (Formulas, Formula Auditing, Evaluate Formula).
Of course, it is language dependent, to `"20.3.2022"*2` works for *cs-CZ* culture, but not *en-US*.

My previous theory was wrong that the problem was in the source cell was incorrect. Problem is with type coercion.

EDIT: Plot thickens, it uses VALUE semantic. e.g. 
* `"($100)" * 2` is -200
* `"7:30:30 AM" * 2` is 0.6256944
* `"1 1/2" * 2` is 3

**Comment by jahav** [comment]
The text coercion was implemented in #1899, based on best guesses of what the "conversions take place are implementation-defined" actually is.
I made a stopgap to fix the autofilter in  #1900 , the autofiler has at least for *this* case, it works, provided:
* I use non-corrupted xlsx. The xlsx from the original post has been corrupted and had to be repaired by Excel and the repaired xlsx was used for the program.
* Set the culture of current thread to something that parses the date

Future works is #1407 (culture from parameter, not current thread) and proper autofilter prerequisite depends on #1894.
